python/GangaLHCb/Lib/GaudiRunApp/GaudiRunAppHandlers.py

```diff
@@ -0,0 +1,346 @@
+
+def genDataFiles(job):
+    """
+    Generating a data.py file which contains the data we want gaudirun to use
+    Args:
+        job (Job): This is the job object which contains everything useful for generating the code
+    """
+    logger.debug("Doing XML Catalog stuff")
+
+    inputsandbox = []
+
+    data = job.inputdata
+    data_str = ''
+    if data:
+        logger.debug("Returning options String")
+        data_str = data.optionsString()
+        if data.hasLFNs():
+            logger.info("Generating Data catalog for job: %s" % job.fqid)
+            logger.debug("Returning Catalogue")
+            inputsandbox.append(FileBuffer('catalog.xml', data.getCatalog()))
+            cat_opts = '\nfrom Gaudi.Configuration import FileCatalog\nFileCatalog().Catalogs = ["xmlcatalog_file:catalog.xml"]\n'
+            data_str += cat_opts
+
+        inputsandbox.append(FileBuffer(data_file, data_str))
+
+    return inputsandbox
+
+def generateWNScript(commandline, job):
+    """
+    Generate the script as a file buffer and return it
+    Args:
+        commandline (str): This is the command-line argument the script is wrapping
+        job (Job): This is the job object which contains everything useful for generating the code
+    """
+    exe_script_name = 'gaudiRun-script.py'
+
+    return FileBuffer(name=exe_script_name, contents=script_generator(gaudiRun_script_template(), COMMAND=commandline,
+                                                                    OUTPUTFILESINJECTEDCODE = getWNCodeForOutputPostprocessing(job, '    ')),
+                      executable=True)
+
+def collectPreparedFiles(app):
+    """
+    Collect the files from the Application in the prepared state
+    Args:
+        app (GaudiRun): This expects only the GaudiRun app
+    """
+    if not isinstance(app.is_prepared, ShareDir):
+        raise ApplicationConfigurationError('Failed to prepare Application Correctly')
+    shared_dir = app.getSharedPath()
+    input_files, input_folders = [], []
+    for root, dirs, files in os.walk(shared_dir, topdown=True):
+        for name in files:
+            input_files.append(os.path.join(root, name))
+        for name in dirs:
+            input_folders.append(os.path.join(root, name))
+    return input_files, input_folders
+
+def prepareCommand(app):
+    """
+    Returns the command which is to be run on the worker node
+    Args:
+        app (GaudiRun): This expects only the GaudiRun app
+    """
+    opts_file = app.getOptsFile()
+    if isinstance(opts_file, (LocalFile, DiracFile)):
+        # Ideally this would NOT need the basename, however LocalFile is special in this regard.
+        # TODO Fix this after fixing LocalFile
+        opts_name = os.path.basename(opts_file.namePattern)
+    else:
+        raise ApplicationConfigurationError("The filetype: %s is not yet supported for use as an opts file.\nPlease contact the Ganga devs is you wish this implemented." %
```

Comment:
Should also be Prepared Error but this isn't handled correctly because we're too specific higher up.